src/Components/Publishing/ReadMore/ReadMoreButton.tsx

```diff
@@ -1,20 +1,34 @@
 import { unica } from "Assets/Fonts"
+import { once } from "lodash"
 import React from "react"
+import Waypoint from "react-waypoint"
 import styled from "styled-components"
 import { track } from "../../../Utils/track"
 import { pMedia } from "../../Helpers"
 import { StandardLayoutParent } from "../Layouts/StandardLayout"
 
-@track()
-export class ReadMore extends React.Component<any, any> {
-  constructor(props) {
-    super(props)
-    this.onClick = this.onClick.bind(this)
+interface ReadMoreProps {
+  onClick: () => void
+  tracking?: any
+}
+
+export class ReadMore extends React.Component<ReadMoreProps, any> {
+  onClick = () => {
+    const { onClick, tracking } = this.props
+
+    tracking.trackEvent({
+      action: "Clicked read more",
+    })
+    onClick()
   }
 
-  @track({ action: "Clicked read more" })
```

Comment:
Can you remind me what we the issues we were getting with using decorators aside from it using the context API and overriding any parent contexts?

Reply:
That was the issue I had problems with in the past, not sure if there are other bugs on different teams-- it will overwrite context, so anything outside of tracking is not shared with children - https://github.com/NYTimes/react-tracking/blob/master/src/withTrackingComponentDecorator.js#L37

Reply:
Got it, because if that's the only problem it may make more sense to fix it upstream in `react-tracking` in a fork and send them a PR instead of removing all decorators as we're using them extensively already.

Reply:
Agreed that this should be fixed upstream, but also find the props method to be more clear and easier to implement, so leaning that way for any new/updated implementations.